docs/DATA.md

```diff
@@ -0,0 +1,94 @@
+
+Different applications can write to the same namespace, but they must agree on what the namespace is for `TODO figure out how to prevent conflicts`.
+
+## Schemas and Sources
+Given the ability for many different applications to write data across many different types of datastores, things will quickly grow to complex for an ecosystem of interoperable applications to thrive.
+
+To support data interoperability, Habitat provides an API for specifying common APIs that communities can implement. This system is split into three components: 
+* Schemas
+* Interfaces
+* Implementations
+
+The purpose of this is to make it easy to tell what APIs a community actually publishes. In a sentence, you might say `"community X implements interface Y which returns schema Z"`. 
+
+`Schemas` are structured data types that are registered with the `data_system_name`. They are content-addressed, so changes to the structure will result in a reference to a fundamentally different schema. They are written in the format of [JSON Schemas](https://json-schema.org/).
+
+`Interfaces` pair a schema with a description of an API being provided. There can be many `Interfaces` with the same `Schema`, as long as their description text differs. The reason for this is that many APIs can serve the same data type while having completely different semantics. For example, if there was a `photo` schema, two very different interfaces for it might be `profile_picture` and `receipt_picture`. The underlying type is the same, but what the API is specifying is completely different.
+
+Each community can maintain a list of `Implementations` for `Interfaces` that it cares about. A `Implementation` is just a path to a specific piece of data following the format `/<community_id>/<datastore>/<namespace>/<content_identifer>` (just the back half of the URL to access any given data in the first place). Note that a community can provide multiple `Implementations` for a single `Interface`. This is to allow different datastores, and different content identifiers within the same store to be able to implement an `Interface`. It is up to applications to choose which `Implementation` to use if multiple are presented. Since not all datastores present their data in a format that converts cleanly to JSON, each datastore must provide a facility to "JSONify" data retrieved via a content identifier. More advanced custom transformations can be defined in the future as well.
+
+Since `Schemas` and `Interfaces` are content-addressed, they are stored by the node, and not by specific communities. `Implementations` are community specific, because they reference paths to data held under specific communities.
```

Comment:
what do you mean the node? which node? are you saying that `schemas` and `interfaces` are stored in the same place where the data is stored?